### ISSUE TITLE
Unable to volume mount named pipe using HostPath on windows container

### ISSUE THREAD

**Issue by atomaras** [kubernetes/kubernetes#67147]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

Trying to create a windows pod that has the host's docker daemon named pipe (\\.\pipe\docker_engine) volume mounted inside it. 
I've made sure the docker version installed on the windows node supports this feature i.e v18.06.0-ce.
This works beautifully when run locally on my Win 10 dev machine using this command:
`docker run -it --rm -v \\.\pipe\docker_engine:\\.\pipe\docker_engine microsoft:windowsservercore:1803 powershell`
But when trying this on K8S using HostPath volume It does not work and fails with this error:

`Error: Error response from daemon: invalid volume specification: 'c:\\.\pipe\docker-engine:c:\\.\pipe\docker-engine'`

Notice that K8S automatically prepends the **C:** which should be culprit making the volume mounting fail.

**What you expected to happen**:

I expect to be able to volume mount named pipes on windows

**How to reproduce it (as minimally and precisely as possible)**:

* Need a 1.11.0 K8S cluster with a Windows Server 1803 node (10.0.17134.1)
See [here](https://docs.microsoft.com/en-us/virtualization/windowscontainers/kubernetes/getting-started-kubernetes-windows) for instructions.
* Install docker and dockerd version **v18.06.0-ce** on the windows node and restart the docker service.
* Create a pod yaml like this and schedule it on k8s
```
apiVersion: v1
kind: Pod
metadata:
  name: docker-pipe-windows
spec:
  nodeSelector:
    beta.kubernetes.io/os: windows
  containers:
    - name: main
      image: micrososft/windowsservercore:1803
      command:
        - powershell
        - Start-Sleep
        - "-s 3600"
      volumeMounts:
        - mountPath: \\.\pipe\docker-engine
          name: docker-pipe
  volumes:
    - name: docker-pipe
      hostPath: 
        path: \\.\pipe\docker-engine
        type: null
  restartPolicy: Never
```
The pod will fail with:

`Error: Error response from daemon: invalid volume specification: 'c:\\.\pipe\docker-engine:c:\\.\pipe\docker-engine'`

If I change the HostPath type to **File** it fails with:

`MountVolume.SetUp failed for volume "docker-pipe" : hostPath type check failed: \\.\pipe\docker-engine is not a file`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
λ  kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-21T09:17:39Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T20:08:34Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
```
λ  kubectl get nodes -o wide
NAME                STATUS    ROLES     AGE       VERSION   EXTERNAL-IP   OS-IMAGE                                      KERNEL-VERSION          CONTAINER-RUNTIME 
warlockk8s-master   Ready     master    37d       v1.11.0   <none>        Red Hat Enterprise Linux Server 7.4 (Maipo)   3.10.0-862.el7.x86_64   docker://1.13.1
warlockk8s-rhel0    Ready     <none>    37d       v1.11.0   <none>        Red Hat Enterprise Linux Server 7.4 (Maipo)   3.10.0-862.el7.x86_64   docker://1.13.1
warlockk8s-rhel1    Ready     <none>    37d       v1.11.0   <none>        Red Hat Enterprise Linux Server 7.4 (Maipo)   3.10.0-862.el7.x86_64   docker://1.13.1
warlockk8s-win0     Ready     <none>    6d        v1.11.0   <none>        Windows Server Standard                       10.0.17134.1        docker://18.6.0
warlockk8s-win1     Ready     <none>    6d        v1.11.0   <none>        Windows Server Datacenter   10.0.17134.1        docker://18.6.0
```
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by PatrickLang** [comment]
There could be a Windows bug here, will need to dig deeper later. Named pipe redirection doesn't work on some builds of Windows when Hyper-V isolation is used. Windows 10 always uses Hyper-V isolation

**Comment by andyzhangx** [comment]
@PatrickLang do you know whether `Windows Server 1803` supports named pipe? On the other hand, k8s will append `c:` if path does not start with a disk drive, I could make the code change there for named pipe.
@atomaras could you provide example about how to create named pipe on windows? thanks.

**Comment by atomaras** [comment]
@andyzhangx Easiest route would be to install Docker for Windows. 
Here is some code also. This is .NET and WCF opening a server that listens on a specific named pipe.

```
using System.ServiceModel;

[ServiceContract(SessionMode = SessionMode.Required)]
    public interface IConsole
    {
        [OperationContract(IsOneWay = true)]
        void Write(string message);

        [OperationContract(IsOneWay = true)]
        void WriteLine(string message);
    }

[ServiceBehavior(InstanceContextMode = InstanceContextMode.Single)]
    public class ConsoleService : IConsole
    {
        public void Write(string message)
        {
            Console.Write(message);
        }

        public void WriteLine(string message)
        {
            Console.WriteLine(message);
        }
    }
class Program
{
        static void Main(string[] args)
        {
           var host = new ServiceHost(new ConsoleService());
            var binding = new NetNamedPipeBinding(NetNamedPipeSecurityMode.Transport)
            {
                OpenTimeout = TimeSpan.FromSeconds(5),
                CloseTimeout = TimeSpan.FromSeconds(5),
                ReceiveTimeout = TimeSpan.MaxValue,
                MaxReceivedMessageSize = int.MaxValue
            };
            host.AddServiceEndpoint(typeof(IConsole), binding, "net.pipe://127.0.0.1/console");
            Console.Write("Opening console named pipe...");
            try
            {
                host.Open();
            }
            catch (Exception ex)
            {
                Console.WriteLine($"failed with error {ex.Message}");
                host.Abort();
                Environment.Exit(-1);
            }
            Console.WriteLine("success");

            host.Closed += (_, __) =>
            {
                Console.WriteLine("Console service closed.");
                Environment.Exit(-1);
            };
            host.Faulted += (_, __) =>
            {
                Console.WriteLine("Console service faulted.");
                Environment.Exit(-1);
            };
        }
}
```

**Comment by andyzhangx** [comment]
Hi @atomaras, have you tried whether docker on windows 1803 supports named pipe?

**Comment by atomaras** [comment]
@andyzhangx 

> I've made sure the docker version installed on the windows node supports this feature i.e v18.06.0-ce.
> This works beautifully when run locally on my Win 10 dev machine using this command:
> docker run -it --rm -v \\.\pipe\docker_engine:\\.\pipe\docker_engine microsoft:windowsservercore:1803 powershell

**Comment by PatrickLang** [comment]
My mistake - this was resolved in #69484.

/close
